apiserver/params/params.go

```diff
@@ -776,13 +776,27 @@ const (
 // It could be a unit, machine or its agent.
 type Status multiwatcher.Status
 
-// TranslateLegacyStatus returns the status value clients expect to see for Juju 1.x.
-func TranslateLegacyStatus(in Status) Status {
+// TranslateLegacyAgentStatus returns the status value clients expect to see for
+// agent-state in versions prior to 1.24
+func TranslateToLegacyAgentState(in Status) Status {
+	// Originally AgentState (a member of api.UnitStatus) could hold one of:
+	// StatusPending
+	// StatusInstalling
+	// StatusStarted
+	// StatusStopped
+	// StatusError
+	// StatusDown
+	// For compatibility reasons we convert modern states (from V2 uniter) into
+	// three of the old ones: StatusPending, StatusStarted or StatusError.
+	// StatusMaintenance can be StatusPending before the start hook has been run
+	// we dont have enough information for that here so we go for started.
 	switch in {
-	case StatusFailed:
-		return StatusDown
-	case StatusActive:
+	case StatusWaiting:
```

Comment:
This isn't correct here - Waiting can occur at any time after a unit is started - it is like Blocked.
The legacy state is Pending prior to the Start hook running.

The fact this method doesn't have enough info to do the required translation means that the method should take more parameters. eg both Agent and Unit status plus whatever else